unit_test/blas/Test_Blas1_team_abs.hpp

```diff
@@ -0,0 +1,292 @@
+#include<gtest/gtest.h>
+#include<Kokkos_Core.hpp>
+#include<Kokkos_Random.hpp>
+#include<KokkosBlas1_team_abs.hpp>
+#include<KokkosBlas1_dot.hpp>
+#include<KokkosKernels_TestUtils.hpp>
+
+namespace Test {
+  template<class ViewTypeA, class ViewTypeB, class Device>
+  void impl_test_team_abs(int N) {
+
+    typedef Kokkos::TeamPolicy<Device>        team_policy ;
+    typedef typename team_policy::member_type team_member ;
+
+    //Launch M teams of the maximum number of threads per team
+    int M = 4;
+    const team_policy policy( M, Kokkos::AUTO );
+    const int team_data_siz = (N%M == 0)?(N/M):(N/M+1);
+
+    typedef typename ViewTypeA::value_type ScalarA;
+    typedef typename ViewTypeB::value_type ScalarB;
+    typedef Kokkos::Details::ArithTraits<ScalarA> AT;
+
+    typedef Kokkos::View<ScalarA*[2],
+       typename std::conditional<
+                std::is_same<typename ViewTypeA::array_layout,Kokkos::LayoutStride>::value,
+                Kokkos::LayoutRight, Kokkos::LayoutLeft>::type,Device> BaseTypeA;
+    typedef Kokkos::View<ScalarB*[2],
+       typename std::conditional<
+                std::is_same<typename ViewTypeB::array_layout,Kokkos::LayoutStride>::value,
+                Kokkos::LayoutRight, Kokkos::LayoutLeft>::type,Device> BaseTypeB;
+
+
+    double eps = std::is_same<ScalarA,float>::value?2*1e-5:1e-7;
+
+    BaseTypeA b_x("X",N);
+    BaseTypeB b_y("Y",N);
+    BaseTypeB b_org_y("Org_Y",N);
+    
+
+    ViewTypeA x = Kokkos::subview(b_x,Kokkos::ALL(),0);
+    ViewTypeB y = Kokkos::subview(b_y,Kokkos::ALL(),0);
+    typename ViewTypeA::const_type c_x = x;
+
+    typename BaseTypeA::HostMirror h_b_x = Kokkos::create_mirror_view(b_x);
+    typename BaseTypeB::HostMirror h_b_y = Kokkos::create_mirror_view(b_y);
+
+    typename ViewTypeA::HostMirror h_x = Kokkos::subview(h_b_x,Kokkos::ALL(),0);
+    typename ViewTypeB::HostMirror h_y = Kokkos::subview(h_b_y,Kokkos::ALL(),0);
+
+    Kokkos::Random_XorShift64_Pool<typename Device::execution_space> rand_pool(13718);
+
+    Kokkos::fill_random(b_x,rand_pool,ScalarA(10));
+    Kokkos::fill_random(b_y,rand_pool,ScalarB(10));
+
+    Kokkos::deep_copy(b_org_y,b_y);
+
+    Kokkos::deep_copy(h_b_x,b_x);
+    Kokkos::deep_copy(h_b_y,b_y);
+
+    ScalarA expected_result = 0;
+    for(int i=0;i<N;i++)
+      expected_result += AT::abs(h_x(i)) * AT::abs(h_x(i));
```

Comment:
This would be a good place to compute a Wilkinson-style running error bound, rather than using an ad-hoc bound (see below).